### PR TITLE
ref(whats-new): Move the logic to update internal seen state

### LINKED ISSUE #77462: Update the logic to set the 'hasSeen' property in What's new

### PR DESCRIPTION
**Pull request by priscilawebdev**
closes https://github.com/getsentry/sentry/issues/77462

Preview of the new behavior as a video/image can tell more than 1.000 words:

https://github.com/user-attachments/assets/6df5ee62-4ec3-4f53-9704-b5683ee9233a

### CODE REVIEW

File: static/app/components/sidebar/broadcasts.tsx [hunk 1]
```diff
@@ -28,15 +28,13 @@ type Props = CommonSidebarProps & {
 
 type State = {
   broadcasts: Broadcast[];
-  error: boolean;
```
Comment (priscilawebdev) [inline static/app/components/sidebar/broadcasts.tsx:31]: noticed that `error` is not being used, so I removed this too. please let me know if that is ok in this PR

File: static/app/components/sidebar/broadcasts.tsx [hunk 2]
```diff
@@ -28,15 +28,77 @@ type Props = CommonSidebarProps & {
...
+  onResetCounter: () => void;
+  whatIsNewRevampFeature: boolean;
+} & Pick<CommonSidebarProps, 'orientation' | 'collapsed' | 'hidePanel'>) {
+  useEffect(() => {
+    return () => {
+      if (whatIsNewRevampFeature) {
+        onResetCounter();
+      }
+    };
+  }, [onResetCounter, whatIsNewRevampFeature]);
```
Comment (priscilawebdev) [inline static/app/components/sidebar/broadcasts.tsx:54]: this is the only new code! basically if the user has access to the feature flag `organizations:what-is-new`, the `hasSeen` state will be reset on unmount. That is the change. The rest is the same

Comment (ArthurKnaus) [inline static/app/components/sidebar/broadcasts.tsx:54]: Nit: Why extract all this into a separate component and not just do it in `componentWillUnmount`?

**Review by ArthurKnaus — APPROVED** [review]

File: static/app/components/sidebar/broadcasts.tsx (hunk 1 above)
Comment (ArthurKnaus) [reply static/app/components/sidebar/broadcasts.tsx:31]: fine for me, but imho we should render a fallback in this case.

File: static/app/components/sidebar/broadcasts.tsx (hunk 2 above)
Comment (ArthurKnaus) [reply static/app/components/sidebar/broadcasts.tsx:54]: nvm, just figured it out. 🙈 We should migrate the class component to a functional one...

Comment (priscilawebdev) [reply static/app/components/sidebar/broadcasts.tsx:54]: yes... i have a code prepared 🙂

**Pull request merged**